src/utils/asyncValidators.js

```diff
@@ -1,48 +1,83 @@
-import web3 from 'public-modules/Utilities/Web3Client';
-import config from 'public-modules/config';
+/* eslint no-throw-literal: 0 */
 
-const sleep = ms => new Promise(resolve => setTimeout(resolve, ms))
+import { actions } from 'layout/App/reducer';
+import { includes } from 'lodash';
 
-function* stubbed() {
-	yield sleep(2000).then(() => '1');
-	yield sleep(2000).then(() => '2');
-	return sleep(2000).then(() => '3');
-}
+const { getTokenBalance } = actions;
 
+const ZERO_BALANCE_ERROR = 'ZERO_BALANCE_ERROR';
+const INSUFFICIENT_BALANCE_ERROR = 'INSUFFICIENT_BALANCE_ERROR';
+const CONTRACT_DOES_NOT_CONFORM_TO_ERC20_ERROR =
+  'CONTRACT_DOES_NOT_CONFORM_TO_ERC20_ERROR';
+const UNKNOWN_ERROR = 'UNKNOWN_ERROR';
```

Comment:
When does the unkown error happen?

Reply:
this would only happen if there was an error besides the contract not conforming to erc20 standard (i.e. network issue or something)